packages/codegen-openapi/__tests__/features/v3/operations.spec.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @jest-environment node
+ */
+
+import { promises as fsPromises } from "fs";
+import * as path from "path";
+
+import { Document } from "../../../src";
+import { getAvailableOperations } from "../../../src/openapi/operations";
+
+describe("Operations", () => {
+  let schema: string;
+  beforeAll(async () => {
+    const file = await fsPromises.readFile(path.resolve(__dirname, "../../schemas/v3/petstore-expanded.json"), "utf8");
+    schema = JSON.parse(file);
+  });
+
+  it("should get available operations for openapi schema", async () => {
+    const operationIdMap = {
+      findPets: {
+        _shouldExist: ["responses", "parameters"],
+        path: "/pets",
+        method: "get",
+      },
+      addPet: {
+        _shouldExist: ["responses", "requestBody"],
+        path: "/pet",
+        method: "post",
+      },
+      deletePet: {
+        _shouldExist: ["parameters", "responses"],
+        path: "/pet/{petId}",
+        method: "delete",
+      },
+    };
+    const operations = getAvailableOperations(schema as unknown as Document);
+    // eslint-disable-next-line no-restricted-syntax
+    for (const operation of operations) {
+      if (!Object.keys(operationIdMap).includes(operation.operationId)) {
+        // eslint-disable-next-line no-continue
+        continue;
```

Comment:
I would assume that this statement should be:
```ts
throw new Error("Object not found")
```

Reply:
No, why? I did not write test for all generated objects and elements, only for the most important ones that cover all the cases.

Reply:
I rewrote the cases so this comment is not applicable anymore.